python/ray/data/_internal/block_batching/iter_batches.py

```diff
@@ -0,0 +1,124 @@
+import collections
```

Comment:
Could use a file-level comment on the overall strategy and pieces.

Reply:
added